charts/ark-solr/templates/statefulset.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "solr.fullname" . }}-statefulset
+  labels:
+    {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: 1
+  selector: 
+    matchLabels:  
+      {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  serviceName: {{ include "solr.fullname" . }}-headless
+  template:
+    metadata:
+      name: ark-solr
+      labels:
+        {{- include "solr-helm-charts.labels" . | nindent 8 }}
+    spec:
+      initContainers:
+      containers:
+      - name: solr
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        ports:
+        - containerPort: 8983
+          name: solr-port
+          protocol: TCP
+        command:
+        livenessProbe:
+        readinessProbe: 
+        resources:
+        volumeMounts:
+        - name: solr-data
+          mountPath: /data
+        - name: solr-conf
+          mountPath: /opt/solr/server/solr/configsets/_default/conf
+      volumes:
+        - name: solr-conf
+          configMap: 
+            name: {{ include "solr.configmap-name" . }}
+        {{- if and .Values.persistence.enabled .Values.persistence.existingClaim }}
+        - name: solr-data
+          persistentVolumeClaim:
+            claimName: {{ printf "%s" (tpl .Values.persistence.existingClaim .) }}
```

Comment:
@naveen-armedia Is there a reason why you are not using `{{ .Values.persistence.existingClaim }}` directly here?

Reply:
@fabricetriboix Not any specific reason. I have seen many community helm charts . And this looks to be more standard when applying existingClaim as it is utilising helm template functions to convert it to string.

Reply:
I would suggest to simply use `{{ .Values.persistence.existingClaim }}` to easy readability and avoid confusing the reader.

Reply:
@fabricetriboix  @JonHolman the standard charts from bitnami and helm stable charts followed in the same manner. Do you want me to change it

Reply:
@naveen-armedia @JonHolman Any comment?

Reply:
@naveen-armedia yes, please try it without.